src/parser/warrior/arms/modules/spells/azeritetraits/ExecutionersPrecision.js

```diff
@@ -1,56 +1,64 @@
 import React from 'react';
 
+import Events from 'parser/core/Events';
+import ExecuteRange from 'parser/warrior/arms/modules/core/Execute/ExecuteRange';
 import { formatPercentage } from 'common/format';
 import SpellLink from 'common/SpellLink';
 import SPELLS from 'common/SPELLS';
-
-import Analyzer from 'parser/core/Analyzer';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
 import Enemies from 'parser/shared/modules/Enemies';
 
 import SpellUsable from '../../features/SpellUsable';
 
 /**
  * Execute increases the damage of your next Mortal Strike against the target by 439, stacking up to 2 times.
+ *
+ * Mortal Strike should be used during the execute phase with this trait while the player have 2 stacks of Executioner's Precision and 2 stacks of Overpower.
  */
 
 class ExecutionersPrecisionAnalyzer extends Analyzer {
   static dependencies = {
     enemies: Enemies,
     spellUsable: SpellUsable,
+    executeRange: ExecuteRange,
   };
 
+  mortalStrikeOnExecPhase = 0;
+  badMortalStrikeOnExecPhase = 0;
+
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTrait(SPELLS.EXECUTIONERS_PRECISION_TRAIT.id);
-  }
 
-  procs = 0;
-  wastedProcs = 0;
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.MORTAL_STRIKE), this._badMortalStrikeCastOnExecPhase);
+  }
 
-  on_byPlayer_cast(event) {
-    if (event.ability.guid !== SPELLS.EXECUTE.id) {
+  _badMortalStrikeCastOnExecPhase(event) {
+    if (!this.executeRange.isTargetInExecuteRange(event)) {
       return;
     }
 
-    this.procs += 1;
+    this.mortalStrikeOnExecPhase += 1;
+
     const enemy = this.enemies.getEntity(event);
-    if (!enemy) {
+    const ep = enemy.getBuff(SPELLS.EXECUTIONERS_PRECISION_DEBUFF.id);
+    const op = this.selectedCombatant.getBuff(SPELLS.OVERPOWER.id);
+    if ((ep === undefined || op === undefined) || (!ep && !op && !enemy)) {
       return;
     }
 
-    const executionersPrecision = enemy.getBuff(SPELLS.EXECUTIONERS_PRECISION_DEBUFF.id);
-    if (executionersPrecision && executionersPrecision.stacks === 2 && this.spellUsable.isAvailable(SPELLS.MORTAL_STRIKE.id)) {
-      this.wastedProcs += 1;
+    if ((ep.stacks !== 2 && op.stacks !== 2) || (ep.stacks !== 2 && op.stacks === 2) || (ep.stacks === 2 && op.stacks !== 2)) {
```

Comment:
couldnt this be simplified by just doing if (ep.stacks !== 2 || op.stacks !== 2) instead of writing out every scenario? That way if EP or OP are not 2 then it is bad?